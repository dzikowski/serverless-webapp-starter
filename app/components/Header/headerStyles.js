export default {
  link: {
    cursor: 'pointer',
    textDecoration: 'none',
    color: 'inherit',
    textTransform: 'uppercase',
    marginLeft: '24px',
  },
};
