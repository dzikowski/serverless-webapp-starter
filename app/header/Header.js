import React, {Component} from "react";
import {Link} from "react-router-dom";
import AppBar from "material-ui/AppBar";
import FlatButton from "material-ui/FlatButton";
import IconMenu from "material-ui/IconMenu";
import MenuItem from "material-ui/MenuItem";
import IconButton from "material-ui/IconButton";
import MoreVertIcon from "material-ui/svg-icons/navigation/more-vert";

const styles = {
  title: {
    cursor: 'pointer',
    'text-decoration': 'none',
    color: 'inherit'
  }
};

const getSignInButton = (handleSignIn) => (
  <FlatButton
    label="Sign In"
    onTouchTap={handleSignIn}
  />
);

const getProfileMenu = (handleSignOut) => (
  <IconMenu
    iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
    targetOrigin={{horizontal: 'right', vertical: 'top'}}
    anchorOrigin={{horizontal: 'right', vertical: 'top'}}
  >
    <MenuItem
      primaryText="Profile"
      containerElement={<Link to="/profile"/>}
    />
    <MenuItem
      primaryText="Sign out"
      onTouchTap={handleSignOut}
    />
  </IconMenu>
);

class Header extends Component {

  state = {
    signedIn: false,
  };

  handleSignIn = () => {
    this.setState({signedIn: true})
  };

  handleSignOut = () => {
    this.setState({signedIn: false});
  };

  render() {
    return (
      <AppBar
        title={<Link to="/" style={styles.title}>Serverless WebApp Starter</Link>}
        onTitleTouchTap={this.goHome}
        iconElementRight={this.state.signedIn ? getProfileMenu(this.handleSignOut) : getSignInButton(this.handleSignIn)}
      />
    );
  };
}

export default Header;
