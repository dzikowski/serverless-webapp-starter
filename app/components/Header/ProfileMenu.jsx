import React from 'react';
import { Link } from 'react-router-dom';
import headerStyles from './headerStyles';

export default function ProfileMenu() {
  return (
    <div>
      <Link style={headerStyles.link} to="/profile/home">Profile</Link>
      <Link style={headerStyles.link} to="/profile/sign-out">Sign out</Link>
      {/*
       Currently we handle this feature with Links, like in case of Register/Sign In. There seems
       to be a bug with icon menu (and menu in the current version of React Toolbox and
       dependencies. Icon menu is open all time and it does not react on any actions. Possible
       relation with https://github.com/react-toolbox/react-toolbox/issues/1453
       <IconMenu icon="more_vert" position="topRight" menuRipple>
       <MenuItem value="profile" caption="Profile" onClick={openProfile} />
       <MenuItem value="signOut" caption="Sign out" onClick={signOut} />
       </IconMenu>
       */}
    </div>
  );
}
