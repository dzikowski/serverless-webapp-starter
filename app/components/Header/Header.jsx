import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';
import { AppBar } from 'react-toolbox/lib/app_bar';
import { Navigation } from 'react-toolbox/lib/navigation';
import { Auth, User } from '../../services/auth';
import ProfileMenu from './ProfileMenu';
import headerStyles from './headerStyles';

function SignInButtons() {
  return (
    <div>
      <Link style={headerStyles.link} to="/profile/register">Register</Link>
      <Link style={headerStyles.link} to="/profile/sign-in">Sign In</Link>
    </div>
  );
}

export default function Header({ auth, user }) {
  return (
    <AppBar title="Serverless WebApp Starter" leftIcon="menu">
      <Navigation type="horizontal">
        {user.signedIn ? <ProfileMenu auth={auth} user={user} /> : <SignInButtons auth={auth} />}
      </Navigation>
    </AppBar>
  );
}

Header.propTypes = {
  auth: PropTypes.instanceOf(Auth).isRequired,
  user: PropTypes.instanceOf(User).isRequired,
};

