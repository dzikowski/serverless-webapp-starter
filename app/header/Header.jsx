import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';
import { AppBar } from 'react-toolbox/lib/app_bar';
import { Navigation } from 'react-toolbox/lib/navigation';
import Auth from '../profile/Auth';
import User from '../profile/User';
import ProfileMenu from './ProfileMenu';

export const headerStyles = {
  link: {
    cursor: 'pointer',
    textDecoration: 'none',
    color: 'inherit',
    textTransform: 'uppercase',
    marginLeft: '24px',
  },
};

function SignInButtons() {
  return (
    <div>
      <Link style={headerStyles.link} to="/profile/register">Register</Link>
      <Link style={headerStyles.link} to="/profile/sign-in">Sign In</Link>
    </div>
  );
}

export default function Header({ auth, user }) {
  return (
    <AppBar title="Serverless WebApp Starter" leftIcon="menu">
      <Navigation type="horizontal">
        {user.signedIn ? <ProfileMenu auth={auth} user={user} /> : <SignInButtons auth={auth} />}
      </Navigation>
    </AppBar>
  );
}

Header.propTypes = {
  auth: PropTypes.instanceOf(Auth).isRequired,
  user: PropTypes.instanceOf(User).isRequired,
};

