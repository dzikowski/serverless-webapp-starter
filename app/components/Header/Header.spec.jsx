import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';
import { User } from '../../services/auth';

it('renders correctly for signed in user', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Header user={User.signedIn('jsparrow@pearl.org')} />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly for signed out user', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Header user={User.signedOut(null)} />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
