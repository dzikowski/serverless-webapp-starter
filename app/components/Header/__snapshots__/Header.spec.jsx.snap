// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for signed in user 1`] = `
<header
  className=""
  data-react-toolbox="app-bar"
>
  <div
    className={undefined}
  >
    <button
      className=" undefined"
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={undefined}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        aria-label=""
        className="material-icons"
        data-react-toolbox="font-icon"
      >
        menu
      </span>
    </button>
    <h1
      className=""
    >
      Serverless WebApp Starter
    </h1>
    <nav
      className=""
      data-react-toolbox="navigation"
    >
      <div>
        <a
          href="/profile/home"
          onClick={[Function]}
          style={
            Object {
              "color": "inherit",
              "cursor": "pointer",
              "marginLeft": "24px",
              "textDecoration": "none",
              "textTransform": "uppercase",
            }
          }
        >
          Profile
        </a>
        <a
          href="/profile/sign-out"
          onClick={[Function]}
          style={
            Object {
              "color": "inherit",
              "cursor": "pointer",
              "marginLeft": "24px",
              "textDecoration": "none",
              "textTransform": "uppercase",
            }
          }
        >
          Sign out
        </a>
      </div>
    </nav>
  </div>
</header>
`;

exports[`renders correctly for signed out user 1`] = `
<header
  className=""
  data-react-toolbox="app-bar"
>
  <div
    className={undefined}
  >
    <button
      className=" undefined"
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={undefined}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <span
        aria-label=""
        className="material-icons"
        data-react-toolbox="font-icon"
      >
        menu
      </span>
    </button>
    <h1
      className=""
    >
      Serverless WebApp Starter
    </h1>
    <nav
      className=""
      data-react-toolbox="navigation"
    >
      <div>
        <a
          href="/profile/register"
          onClick={[Function]}
          style={
            Object {
              "color": "inherit",
              "cursor": "pointer",
              "marginLeft": "24px",
              "textDecoration": "none",
              "textTransform": "uppercase",
            }
          }
        >
          Register
        </a>
        <a
          href="/profile/sign-in"
          onClick={[Function]}
          style={
            Object {
              "color": "inherit",
              "cursor": "pointer",
              "marginLeft": "24px",
              "textDecoration": "none",
              "textTransform": "uppercase",
            }
          }
        >
          Sign In
        </a>
      </div>
    </nav>
  </div>
</header>
`;
